### PR TITLE
Add visual alert for Ticket Task is_tickettaskcategory_mandatory

### PR DESCRIPTION
**Pull request by tsmr**
image

### CODE REVIEW

**Comment by tsmr** [comment]
ping :) ?

**Review by yllen — CHANGES_REQUESTED** [review]
why this message in the itilsolution class?
you have a class for tickettask

**Comment by tsmr** [comment]
Because i believe than you can add one class in the hook :

`$PLUGIN_HOOKS['pre_item_form']['behaviors'] = [PluginBehaviorsITILSolution::class, 'messageWarningSolution'];`

**Comment by yllen** [comment]
If you can only add one of this type of class in the setup, move this in the common.class

**Comment by tsmr** [comment]
I move 3 functions to ticket class

File: inc/common.class.php [hunk 1]
```diff
@@ -211,4 +211,192 @@ static function cloneItem(Array $param) {
...
+               echo "<div class='d-flex'>";
+
+               echo "<div class='me-2'>";
+               echo "<i class='fa fa-exclamation-triangle fa-2x'></i>";
+               echo "</div>";
+
+               echo "<div>";
+               echo "<h4 class='alert-title'>" . __('You cannot resolve the ticket', 'behaviors') . "</h4>";
+               echo "<div class='text-muted'>" . implode('</div><div>', $warnings) . "</div>";
+               echo "</div>";
```
Comment (yllen) [inline inc/common.class.php:348]: missing part for ticket solutiontype 
(i can't resolve the conflict)

**Review by yllen — CHANGES_REQUESTED** [review]

File: inc/common.class.php (hunk 1 above)
Comment (tsmr) [reply inc/common.class.php:348]: I don't understand why thuis part has missing ? it's ok for me. For conflict, i can revert changes for inc/itilsolution.class.php temporarily ?

**Comment by tsmr** [comment]
It's ok now. In my next PR il will delete old functions in itilsolution.class.php

**Pull request merged**